### PR TITLE
Fixes #2233 Make Publication citation style not dependent on View Mode

### LINKED ISSUE #2233: Make Publication citation style not dependent on View Mode
## Motivation
Currently, selection of citation style depends on publication node rendering view mode. This isn't desirable for long-term maintainability as it requires views have one display to support each citation style, and it also makes it more difficult to add new citation styles to a quickstart site (as citation style is linked to view mode)

## Proposed Resolution
Selection of citation style when a publication is displayed should not depend on view mode.

### Describe the solution you'd like
- Unlink publication view mode display from citation style
- Add a solution that allows a default citation style to be selected for a site
- Allows views to specify a citation style in a way that does not involve adding more displays to the view

### Describe alternatives you've considered
Keeping the conceptual link between node view mode and citation style.

### PR DESCRIPTION
**Pull request by tadean**
## Description
Currently, publication citations are dependent on publication view mode. This works, but results in many extra view displays being needed to control rendering in these view modes, which is not ideal from a maintainability standpoint. It also makes it more difficult to add new citation styles to the site.

This PR adds a configuration settings form to set the site default citation style, and a special views argument `az_citation_style_argument` handler that selects citation style. This means that citation style for views can be made to vary from the site default by passing an argument to the view.

**Note: the changes in this PR are based upon the changes in #2231 and it is best reviewed compared to that branch.**

## Related issues
#2233 

## How to test

- enable `az_publication`
- create some publications
- select different citation styles on the publication settings form at `/admin/config/az-quickstart/settings/az-publication`
- verify that publications are rendered in the chosen citation style
- create a flexible page with a view reference added to `AZ Publications`. Set the view argument to the name of a citation style, e.g. `az_mla` or `az_ieee`, etc. Citation styles are listed at `/admin/structure/az_citation_style`
- verify that the view reference renders in the selected citation style.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [x] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [x] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_publication/az_publication.module [hunk 1]
```diff
@@ -16,6 +16,7 @@ use Drupal\Core\Url;
 use Drupal\Core\Link;
 use Symfony\Component\HttpFoundation\Request;
 use Drupal\node\NodeInterface;
+use Drupal\az_publication\Plugin\views\argument\AZCitationStyleArgument;
```
Comment (trackleft) [inline modules/custom/az_publication/az_publication.module:19]: woah!

**Review by trackleft — DISMISSED** [review]
Appears to work as expected, nice work!  We should add a list of available arguments in our Documentation
```
az_bluebook
az_ama
az_apa
az_chicago
az_ieee
az_mla
```

**Comment by joeparsons** [comment]
Link for comparing code in this PR to #2231 PR branch
https://github.com/az-digital/az_quickstart/compare/feature/2214...feature/2233

**Comment by trackleft** [comment]
Question, is it possible to add the contextual filter introduced in this PR to other views, and if someone were to do that, would it cause an error?

**Review by bberndt-uaz — DISMISSED** [review]

**Comment by tadean** [comment]
Updates:
- Relocates citation style settings to a sub-tab of the quickstart publication settings page (thanks @joeparsons for suggestion)
- Add actual links to the CSL project on the tooltip for the style edit form text area
- Merge person block view into the main publication view; with the removal of the unneeded view modes it's not necessary they be separated
- Add template for flexible block version of the publications person block so that if a publications person block needs to be created with an argument (e.g. specified style) it comes with the correct padding for the person page

**Review by trackleft — DISMISSED** [review]

**Review by bberndt-uaz — DISMISSED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**